Feature/query wizard

ca1yps0 opened the pull request.

ca1yps0 committed:
feat(query-builder): implement ember-cli-wizard

ca1yps0 committed:
fix: ESLint
Merge branch 'develop' into feature/query-wizard

# Conflicts:
#	app/components/elvis-select-guru.js
#	app/templates/network/new.hbs
#	package-lock.json

ca1yps0 committed:
fix: eslint

nightsh commented:
- steps line is not aligned properly (1-2 px to the right should do)
- "loading" bits are not set on wizard step change (hence a slower "loading" feedback)
- step titles look like anchors, they shouldn't
- settings: images don't have correct paths
- settings drop downs styling

victor-nitu reviewed (changes requested):
- [x] you should receive an error message if you didn't select anything
  - [x] if nothing was selected in step 1, allow step 2 but don't allow step 3 if nothign was selected
- [x] if you select a company, then you go back and change countries, when you go back to actors the old selection is still there (make it disappear if you change any previous steps)
- [x] steps line is not aligned properly (1-2 px to the right should do)
- [x] "loading" bits are not set on wizard step change (hence a slower "loading" feedback)
- [x] step titles look like anchors, they shouldn't
- [x] settings: images don't have correct paths
- [x] settings drop downs styling

Commit:
refactor(query-builder/actor-step): restyle the actor selector

- create separate component for actor selection
- use smarter styles to render network actor figure circles
- better styles for ember select guru selected values and options

Commit:
fix(network-entity): fix method for generating network entity figure

Commit:
chore(legend): delete legend from the query builder

Commit:
refactor(actor-select): improved actor select UX

Commit:
refactor(actor-select): improve coloring and padding

- improve box loading design

victor-nitu committed:
fix: package-lock

victor-nitu committed:
fix: add vis to package.json

ca1yps0 committed:
feat: wizard title selected options

* fix eslint
* add error message
* style adjustments
* remove queryterm after finished selecting actors

ca1yps0 committed:
feat:  do not update years or cpvs if is not necessarily

ca1yps0 committed:
Merge branch 'feature/query-wizard' of github.com:tenders-exposed/elvis-ember into feature/query-wizard

ca1yps0 committed:
Merge branch 'feature/query-wizard' of github.com:tenders-exposed/elvis-ember into feature/query-wizard

# Conflicts:
#	app/controllers/network/new.js
#	app/styles/components/wizard.scss
#	app/templates/network/partials/wizard-actors.hbs

ca1yps0 committed:
fix: wizard title completed steps

ca1yps0 committed:
Merge branch 'feature/query-wizard' of github.com:tenders-exposed/elvis-ember into feature/query-wizard

# Conflicts:
#	app/controllers/network/new.js

ca1yps0 committed:
fix: cleaning code

victor-nitu committed:
Merge branch 'develop' into feature/query-wizard

victor-nitu committed:
fix: lint

victor-nitu commented on app/components/actor-select-guru.js:
```diff
@@ -0,0 +1,88 @@
+export default EmberSelectGuru.extend({
+  // checkActors: false,
+  checkboxIsSelected: false,
+  /*selectAllOptions: observer('checkboxIsSelected', function() {
```
dead code

victor-nitu commented on app/components/actor-select-guru.js:
```diff
@@ -0,0 +1,88 @@
+import { get } from '@ember/object';
+
+export default EmberSelectGuru.extend({
+  // checkActors: false,
```
dead code

victor-nitu commented on app/components/actor-select-guru.js:
```diff
@@ -0,0 +1,88 @@
+      });
+      let onSelect = get(this, 'onSelect');
+      onSelect(this.get('_value'));
+      // this.attrs.onSelect(this.get('_value'));
```
more dead code

victor-nitu reviewed (changes requested):
Not going to go through all the diff, please remove all dead code before anything.

victor-nitu committed:
chore: remove commented out code

zufanka commented:
Great job guys. @ca1yps0 @nightsh @chutko 
There are still a couple of things to fix:

**BUGS**
- [ ] if you don't select a CPV, the "Settings" step is empty
- [ ] if you go to "Create a network" from the network view (back to the query builder), the previous countries are actually still selected, but you don't see it in the first step (if you go a step further, the flags appear). Everything should be deselected if you do this
- [ ] when I press "Launch the network", a loader should appear
- [ ] flags missing here
![image](https://user-images.githubusercontent.com/2411066/56694788-ca5bc080-66e7-11e9-87a2-a06c09647cba.png)
- [ ] This errormessage appears when server returns 500. This should NOT be the case as it's NOT the problem. Can we change it to "This should not be happening. Please send us an email at tech@tenders.exposed if it is." or something like that?
![image](https://user-images.githubusercontent.com/2411066/56694959-3ccca080-66e8-11e9-9f4d-979af5a47ff5.png)


**STYLING**
- [ ] this selection goes into the footer. Please give the element the same height all the steps in the the whole wizard
![image](https://user-images.githubusercontent.com/2411066/56692515-a649b080-66e2-11e9-8f01-45865be27adc.png)
- [ ] there should be more space between these
![image](https://user-images.githubusercontent.com/2411066/56694220-7f8d7900-66e6-11e9-8eaf-fedc710a7806.png)
- [ ] there should be definitely more space here
![image](https://user-images.githubusercontent.com/2411066/56694285-a21f9200-66e6-11e9-8341-a3ce66a1fb46.png)
- [ ] the "Node types" and "edge types" as well as the "name of the network" are a bit small and difficult to read. @chutko can we make them pop out more?
![image](https://user-images.githubusercontent.com/2411066/56694365-c5e2d800-66e6-11e9-8520-a026dcbf65b2.png)
- [ ] can we have these on a transparent background? @chutko 
![image](https://user-images.githubusercontent.com/2411066/56694398-d5fab780-66e6-11e9-87cc-20cca19522ca.png)
- [ ] There should be some indicator that the user chose actors here (no countries chosen now)
![image](https://user-images.githubusercontent.com/2411066/56694859-0000a980-66e8-11e9-9d3a-d04b9165bb2b.png)

victor-nitu commented:
@zufanka I will merge and copy your message into a new issue - we will surely rebranch after merging but this is getting again and again out of sync with develop and it's becoming a hassle to maintain it open & mergeable for so long.
